### PR TITLE
Support model automatic detection

### PR DESCRIPTION
**Pull request by yushijinhun**
* `SkinViewer.slim` is removed, the model is automatically detected by default.
* Add `SkinViewer.detectModel` property, it can be used to turn off the automatic detection feature.
  * Use `SkinViewer.playerObject.skin.slim` property to specify the model manually.
* Export `isSlimSkin` function to allow others to detect the model without creating a `SkinViewer`.

### CODE REVIEW

File: types/utils.d.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export function isSlimSkin(skinImage: HTMLImageElement): boolean;
```
Comment (g-plane) [inline types/utils.d.ts]: Parameter type should be `HTMLImageElement | HTMLCanvasElement`.

Comment (g-plane) [reply types/utils.d.ts]: Missing defs of `loadSkinToCanvas` and `loadCapeToCanvas`.

**Comment by g-plane** [comment]
Tests should be added.

File: types/utils.d.ts (hunk 1 above)
Comment (yushijinhun) [reply types/utils.d.ts]: @g-plane I don't think `HTMLCanvasElement` should be acceptable. It's only for internal use. The canvas is required to have the skin image drawn on it and needs preprocessing, which is complex. And we haven't exported interfaces for that.

**Comment by yushijinhun** [comment]
@g-plane Unit test environment hasn't been set up in the repository yet. Could you do this in a new PR? If so, I will add tests for the feature after your PR is merged.

**Pull request merged**